### ISSUE TITLE
Wrong default Server Name in CSIDE

### ISSUE THREAD

**Issue by CRetsch** [microsoft/navcontainerhelper#997]
When creating a container with the settings below and attempting to open the database in CSIDE, there is by default a wrong Server Name: "xyz\SQLEXPRESS". This causes an error because the server cannot be found. 
The "\SQLEXPRESS" part needs to be removed manually every time we are opening CSIDE, or, for a more permanent solution, the path of the start menu shortcut needs to be adjusted. 

Windows and Web Client are working perfectly well. 
 

New-NavContainer 
-accept_outdated 
-accept_eula 
-alwaysPull 
-containerName $containerName 
-imageName $imageName 
-auth NavUserPassword 
-licensefile $mylicense 
-bakFile "C:\Temp\database.bak" 
-updateHosts 
-includeCSide 
-clickonce 
-assignPremiumPlan 
-shortcuts CommonStartMenu 
-doNotExportObjectsToText 
-enableSymbolLoading 
-memoryLimit "8g" 
-myScripts @("C:\Temp\_ContainerScriptOverrides\SetupWindowsUsers.ps1") 
-additionalParameters $additionalParameters 
-Credential $credential
```

NavContainerHelper is version 0.6.5.11
NavContainerHelper is running as administrator
Container OS Version: 10.0.17763.678
Container NAV Version: 14.5.35970.0-de
OnPrem
```

This happens with all created containers.

**Comment by freddydk** [comment]
When I create a NAV 2018 container, the shortcut is:
```
"C:\ProgramData\NavContainerHelper\Extensions\test\Program Files\110\RoleTailored Client\finsql.exe" servername=test\SQLEXPRESS, Database=CRONUS, ntauthentication=0, ID=test
```
and that works, since the SQL server in the container is using instance name SQLEXPRESS.
I would need the full output of your new-navcontainer to understand why this doesn't work for you.

**Comment by freddydk** [comment]
What error are you getting when launching CSIDE?

**Comment by CRetsch** [comment]
This is the error message we get when trying to launch CSIDE:

![CSIDE_error](https://user-images.githubusercontent.com/65768320/83233013-5ca58200-a18e-11ea-978b-6116e4cdd209.PNG)


Here is the complete output of the new container:


```
NavContainerHelper is version 0.6.5.11
NavContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
Docker Client Version is 19.03.5
Docker Server Version is 19.03.5
Pulling image mcr.microsoft.com/businesscentral/onprem:14.5.35970.0-de-ltsc2019
14.5.35970.0-de-ltsc2019: Pulling from businesscentral/onprem
Using image mcr.microsoft.com/businesscentral/onprem:14.5.35970.0-de-ltsc2019
Removing CR19 from host hosts file
Removing C:\ProgramData\NavContainerHelper\Extensions\CR19
Creating Container CR19
Version: 14.5.35970.0-de
Style: onprem
Platform: 14.0.35916.0
Generic Tag: 0.0.9.94
Container OS Version: 10.0.17763.678 (ltsc2019)
Host OS Version: 10.0.17763.1217 (ltsc2019)
WARNING: The container operating system matches the host operating system, but the revision is different.
If you encounter issues, you might want to specify -isolation hyperv
Using locale de-DE
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\Temp\Entwickler_365BC 14.x (60 Named User, 25 Team Member).flf
Files in C:\ProgramData\NavContainerHelper\Extensions\CR19\my:
- AdditionalOutput.ps1
- AdditionalSetup.ps1
- license.flf
- MainLoop.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- SetupWebClient.ps1
- SetupWindowsUsers.ps1
- updatehosts.ps1
Creating container CR19 from image mcr.microsoft.com/businesscentral/onprem:14.5.35970.0-de-ltsc2019
7d37463cf84c376583ce526430685979af841574a1686b50275c39cb6f1df8c8
Waiting for container CR19 to be ready
Initializing...
Setting host.containerhelper.internal to 172.27.32.1 in container hosts file
Starting Container
Hostname is CR19
PublicDnsName is CR19
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Using Database .bak file 'C:\ProgramData\NavContainerHelper\Extensions\CR19\database.bak'
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 6995AC00AE8BA67888F1461F3C4F44606DA4A83D
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Creating Windows user bsh
Setting SA Password and enabling SA
Creating bsh as SQL User and add to sysadmin
Assign Premium plan
Creating ClickOnce Manifest
Container IP Address: 172.27.43.133
Container Hostname  : CR19
Container Dns Name  : CR19
Web Client          : http://CR19/NAV/
Dev. Server         : http://CR19
Dev. ServerInstance : NAV
ClickOnce Manifest  : http://CR19:8080/NAV
Setting CR19 to 172.27.43.133 in host hosts file
```

**Comment by freddydk** [comment]
What do you have in additionalparameters and setupWindowsUsers?

You have a lot of files in the override section:
- AdditionalOutput.ps1
- AdditionalSetup.ps1
- license.flf
- MainLoop.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- SetupWebClient.ps1
- SetupWindowsUsers.ps1
- updatehosts.ps1

Can you confirm that this is a problem on your machine with standard script:

```
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)
New-NavContainer `
    -accept_outdated `
    -accept_eula `
    -containerName "test" `
    -imageName "mcr.microsoft.com/businesscentral/onprem:14.5.35970.0-de-ltsc2019" `
    -auth NavUserPassword `
    -updateHosts `
    -includeCSide `
    -doNotExportObjectsToText `
    -memoryLimit "8g" `
    -Credential $credential
```
And is the problem when you are using the clickonce deployed CSIDE or the shortcut?

**Comment by FSharpCSharp** [comment]
So I can only join @CRetsch. With me exactly the same behaviour occurs, and I can also reproduce it without hesitation. I always have to correct the link first or change the database manually in the C/SIDE environment. Otherwise I get exactly the same error here.

**Comment by freddydk** [comment]
I trust you, but I cannot repro.
And the strange thing is, that in the container, the SQL instance is called SQLEXPRESS.
Are you using clickonce or the shortcut?
have you tried to delete .zup files on the host?

**Comment by freddydk** [comment]
The reason for this is likely that UDP on 1434 is disallowed on the network or on the host.
I have fixed the CSIDE shortcut to avoid the instance if running localhost\SQLEXPRESS, assuming that this is a single instance installation then.

**Comment by freddydk** [comment]
Shipped in 0.7.0.6
